lib/node-metrics.js

```diff
@@ -0,0 +1,83 @@
+// @ts-check
+const AWS = require('aws-sdk');
+const loopbench = require('loopbench')();
+
+const logger = require('./logger');
+const config = require('./config');
+
+let intervalId;
+
+function emit() {
+  if (!config.runningInEc2) return;
+
+  const memoryStats = process.memoryUsage();
+  const cpuStats = process.cpuUsage();
+  const metrics = [
+    {
+      MetricName: 'NodeEventLoopDelay',
+      Unit: 'Milliseconds',
+      Value: loopbench.delay,
+    },
+    {
+      MetricName: 'NodeMemoryRss',
+      Unit: 'Bytes',
+      Value: memoryStats.rss,
+    },
+    {
+      MetricName: 'NodeMemoryHeapTotal',
+      Unit: 'Bytes',
+      Value: memoryStats.heapTotal,
+    },
+    {
+      MetricName: 'NodeMemoryHeapUsed',
+      Unit: 'Bytes',
+      Value: memoryStats.heapUsed,
+    },
+    {
+      MetricName: 'NodeMemoryExternal',
+      Unit: 'Bytes',
+      Value: memoryStats.external,
+    },
+    {
+      MetricName: 'NodeCpuUser',
+      Unit: 'Microseconds',
+      Value: cpuStats.user,
+    },
+    {
+      MetricName: 'NodeCpuSystem',
+      Unit: 'Microseconds',
+      Value: cpuStats.system,
+    },
```

Comment:
I'm not sure how useful these are to us. They give us microsecond values, which I don't really have a good intuition for. Perhaps this will just be valuable to show us when CPU usage spikes? i.e., the absolute values aren't too useful, but if we see CPU usage double, we might have a problem?

Reply:
If we pass the previous value then I think we can get the increment: https://nodejs.org/api/process.html#processcpuusagepreviousvalue

We could then divide by the 5 second interval to get a percentage, which might be more useful?

Reply:
It might also be helpful to separately report the total CPU% (user + system)?